airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceSelectQueryGeneratorTest.kt

```diff
@@ -0,0 +1,163 @@
+/* Copyright (c) 2024 Airbyte, Inc., all rights reserved. */
+package io.airbyte.integrations.source.mysql
+
+import com.fasterxml.jackson.databind.JsonNode
+import io.airbyte.cdk.discover.Field
+import io.airbyte.cdk.jdbc.DoubleFieldType
+import io.airbyte.cdk.jdbc.IntFieldType
+import io.airbyte.cdk.jdbc.LongFieldType
+import io.airbyte.cdk.jdbc.LosslessJdbcFieldType
+import io.airbyte.cdk.jdbc.OffsetDateTimeFieldType
+import io.airbyte.cdk.jdbc.StringFieldType
+import io.airbyte.cdk.read.And
+import io.airbyte.cdk.read.Equal
+import io.airbyte.cdk.read.From
+import io.airbyte.cdk.read.FromSample
+import io.airbyte.cdk.read.Greater
+import io.airbyte.cdk.read.LesserOrEqual
+import io.airbyte.cdk.read.Limit
+import io.airbyte.cdk.read.Or
+import io.airbyte.cdk.read.OrderBy
+import io.airbyte.cdk.read.SelectColumnMaxValue
+import io.airbyte.cdk.read.SelectColumns
+import io.airbyte.cdk.read.SelectQuery
+import io.airbyte.cdk.read.SelectQuerySpec
+import io.airbyte.cdk.read.Where
+import io.airbyte.cdk.read.optimize
+import io.airbyte.cdk.util.Jsons
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+class MysqlSourceSelectQueryGeneratorTest {
+    @Test
+    fun testSelectLimit0() {
+        SelectQuerySpec(
+                SelectColumns(
+                    listOf(
+                        Field("k", IntFieldType),
+                        Field("v", StringFieldType),
+                    ),
+                ),
+                From("TBL", "SC"),
+                limit = Limit(0),
+            )
+            .assertSqlEquals("""SELECT `k`, `v` FROM `SC`.`TBL` LIMIT 0""")
+    }
+
+    @Test
+    fun testSelectMaxCursor() {
+        SelectQuerySpec(
+                SelectColumnMaxValue(Field("ts", OffsetDateTimeFieldType)),
+                From("TBL", "SC"),
+            )
+            .assertSqlEquals("""SELECT MAX(`ts`) FROM `SC`.`TBL`""")
+    }
+
+    @Test
+    fun testSelectSample() {
+        SelectQuerySpec(
+                SelectColumns(
+                    listOf(
+                        Field("k", IntFieldType),
+                        Field("v", StringFieldType),
+                    ),
+                ),
+                FromSample("TBL", "SC", 3, 1000),
+                orderBy = OrderBy(listOf(Field("k", IntFieldType))),
+            )
+            .assertSqlEquals(
+                """SELECT `k`, `v`""" + """ FROM `SC`.`TBL` limit 1000 ORDER BY `k`""",
```

Comment:
Is this a valid sql query in mysql?

Reply:
good catch. fixed - we don't need that limit anyway.